.github/workflows/build_cmake_tarball.yml

```diff
@@ -0,0 +1,165 @@
+name: CMake Tarball
+
+#  This file is part of t8code.
+#  t8code is a C library to manage a collection (a forest) of multiple
+#  connected adaptive space-trees of general element types in parallel.
+#
+#  Copyright (C) 2024 the developers
+#
+#  t8code is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  t8code is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with t8code; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+#
+# This github CI script constructs a tarball and checks if it is build correctly.
+#
+
+env:
+  MAKEFLAGS: "-j2 V=0"
+
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+      - CI-*tarball* # for testing this script, all feature branches with "tarball" in their name
+  pull_request:
+    branches:
+      - main
+      - develop
+  workflow_dispatch: 
```

Comment:
please put this workflow to the other cmake workflows just like the t8code_w_shipped_submodules test